packages/recoil/recoil_values/Recoil_selector.js

```diff
@@ -700,7 +700,7 @@ function selector<T>(
   ): void {
     deps.add(newDepKey);
 
-    setDepsInStore(store, state, deps, executionId);
+    // setDepsInStore(store, state, deps, executionId);
```

Comment:
With this simplification we can just get rid of `setNewDepInStore()` and call `deps.add()` directly in `getRecoilValue()`.